flags/tests/test_state.py

```diff
@@ -22,8 +22,22 @@ def test_flag_state_disabled(self):
 
     def test_flag_state_non_existent_flag_site(self):
         """ Given a site non-existent flags should still be False """
+        request = self.factory.get('/test')
         self.assertFalse(flag_state('FLAG_DOES_NOT_EXIST',
-                                    request=self.request))
+                                    request=request))
+
+    def test_flag_state_request_in_kwargs_with_conditions(self):
```

Comment:
I wonder if these tests would rather be in test_middleware.py, to avoid duplicating the logic that the middleware performs. See [these two tests](https://github.com/cfpb/django-flags/compare/flags-middleware...chosak:flags-middleware-review#diff-9b77c8e96d92d2d495dd5031bdfd94d5R43) for a proposed implementation that verifies that the middleware caches the flags.